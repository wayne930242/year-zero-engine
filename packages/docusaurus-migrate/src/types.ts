/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export type RawData = {
  header?: string;
  content?: string;
};

export type Data = {
  metadata: {[key: string]: string};
  rawContent: string;
};

export type ClassicPresetEntries = {
  docs: {[key: string]: any};
  blog: {[key: string]: any};
  theme: {[key: string]: any};
};

export type SidebarEntries = {
  [key: string]:
    | Record<string, unknown>
    | Array<Record<string, unknown> | string>;
};

export interface VersionTwoConfig {
  baseUrl: string;
  favicon: string;
  tagline?: string;
  title: string;
  url: string;
  organizationName?: string;
  projectName?: string;
  noIndex?: boolean;
  githubHost?: string;
  onBrokenLinks: string;
  onBrokenMarkdownLinks: string;
  plugins: Array<[string, {[key: string]: any}]>;
  themes?: [];
  presets: [[string, ClassicPresetEntries]];
  themeConfig: {
    gtag?: {
      trackingID?: string;
    };
    navbar: {
      title?: string;
      logo?: {
        src?: string;
      };
      items: Array<Record<string, unknown> | null>;
    };
    image?: string;
    footer: {
      links: Array<{
        title: string;
        items: Array<{
          label: string;
          to: string;
        }>;
      }>;
      copyright?: string;
      logo: {
        src?: string;
      };
    };
    algolia?: Record<string, unknown>;
  };
  customFields: {
    [key: string]: unknown;
  };
  scripts?: (
    | string
    | {
        src: string;
        [key: string]: unknown;
      }
  )[];
  stylesheets?: (
    | string
    | {
        href: string;
        [key: string]: unknown;
      }
  )[];
}

export type VersionOneConfig = {
  title?: string;
  tagline?: string;
  url?: string;
  baseUrl?: string;
  defaultVersionShown?: string;
  organizationName?: string;
  projectName?: string;
  noIndex?: boolean;
  headerLinks?: Array<any>;
  headerIcon?: string;
  favicon?: string;
  colors?: any;
  copyright?: string;
  editUrl?: string;
  customDocsPath?: string;
  users?: Array<Record<string, unknown>>;
  disableHeaderTitle?: string;
  disableTitleTagline?: string;
  separateCss?: Array<Record<string, unknown>>;
  footerIcon?: string;
  translationRecruitingLink?: string;
  algolia?: Record<string, unknown>;
  gaTrackingId?: string;
  highlight?: Record<string, unknown>;
  markdownPlugins?: Array<() => void>;
  scripts?: Array<{src: string; [key: string]: any} | string>;
  stylesheets?: Array<{href: string; [key: string]: any} | string>;
  facebookAppId?: string;
  facebookComments?: true;
  facebookPixelId?: string;
  twitter?: string;
  twitterUsername?: string;
  twitterImage?: string;
  ogImage?: string;
  cleanUrl?: boolean;
  scrollToTop?: boolean;
  scrollToTopOptions?: Record<string, unknown>;
};
