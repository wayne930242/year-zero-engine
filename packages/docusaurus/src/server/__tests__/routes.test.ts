/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import loadRoutes from '../routes';
import {RouteConfig} from '@docusaurus/types';

describe('loadRoutes', () => {
  test('nested route config', async () => {
    const nestedRouteConfig: RouteConfig = {
      component: '@theme/DocPage',
      path: '/docs:route',
      modules: {
        docsMetadata: 'docs-b5f.json',
      },
      routes: [
        {
          path: '/docs/hello',
          component: '@theme/DocItem',
          exact: true,
          modules: {
            content: 'docs/hello.md',
            metadata: 'docs-hello-da2.json',
          },
        },
        {
          path: 'docs/foo/baz',
          component: '@theme/DocItem',
          modules: {
            content: 'docs/foo/baz.md',
            metadata: 'docs-foo-baz-dd9.json',
          },
        },
      ],
    };
    const result = await loadRoutes([nestedRouteConfig], '/');
    expect(result).toMatchSnapshot();
  });

  test('flat route config', async () => {
    const flatRouteConfig: RouteConfig = {
      path: '/blog',
      component: '@theme/BlogListPage',
      exact: true,
      modules: {
        items: [
          {
            content: {
              __import: true,
              path: 'blog/2018-12-14-Happy-First-Birthday-Slash.md',
              query: {
                truncated: true,
              },
            },
            metadata: 'blog-2018-12-14-happy-first-birthday-slash-d2c.json',
          },
          {
            content: 'blog/2018-12-14-Happy-First-Birthday-Slash.md',
            metadata: null,
          },
        ],
      },
    };
    const result = await loadRoutes([flatRouteConfig], '/');
    expect(result).toMatchSnapshot();
  });

  test('invalid route config', async () => {
    const routeConfigWithoutPath = {
      component: 'hello/world.js',
    } as RouteConfig;

    expect(loadRoutes([routeConfigWithoutPath], '/')).rejects
      .toMatchInlineSnapshot(`
      [Error: Invalid routeConfig (Path must be a string and component is required) 
      {"component":"hello/world.js"}]
    `);

    const routeConfigWithoutComponent = {
      path: '/hello/world',
    } as RouteConfig;

    expect(loadRoutes([routeConfigWithoutComponent], '/')).rejects
      .toMatchInlineSnapshot(`
      [Error: Invalid routeConfig (Path must be a string and component is required) 
      {"path":"/hello/world"}]
    `);
  });

  test('route config with empty (but valid) path string', async () => {
    const routeConfig = {
      path: '',
      component: 'hello/world.js',
    } as RouteConfig;

    const result = await loadRoutes([routeConfig], '/');
    expect(result).toMatchSnapshot();
  });
});
