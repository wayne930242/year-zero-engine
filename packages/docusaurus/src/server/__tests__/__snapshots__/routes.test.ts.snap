// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadRoutes flat route config 1`] = `
Object {
  "registry": Object {
    "component---theme-blog-list-pagea-6-a-7ba": Object {
      "loader": "() => import(/* webpackChunkName: 'component---theme-blog-list-pagea-6-a-7ba' */ \\"@theme/BlogListPage\\")",
      "modulePath": "@theme/BlogListPage",
    },
    "content---blog-0-b-4-09e": Object {
      "loader": "() => import(/* webpackChunkName: 'content---blog-0-b-4-09e' */ \\"blog/2018-12-14-Happy-First-Birthday-Slash.md?truncated=true\\")",
      "modulePath": "blog/2018-12-14-Happy-First-Birthday-Slash.md?truncated=true",
    },
    "content---blog-7-b-8-fd9": Object {
      "loader": "() => import(/* webpackChunkName: 'content---blog-7-b-8-fd9' */ \\"blog/2018-12-14-Happy-First-Birthday-Slash.md\\")",
      "modulePath": "blog/2018-12-14-Happy-First-Birthday-Slash.md",
    },
    "metadata---blog-0-b-6-74c": Object {
      "loader": "() => import(/* webpackChunkName: 'metadata---blog-0-b-6-74c' */ \\"blog-2018-12-14-happy-first-birthday-slash-d2c.json\\")",
      "modulePath": "blog-2018-12-14-happy-first-birthday-slash-d2c.json",
    },
  },
  "routesChunkNames": Object {
    "/blog-94e": Object {
      "component": "component---theme-blog-list-pagea-6-a-7ba",
      "items": Array [
        Object {
          "content": "content---blog-0-b-4-09e",
          "metadata": "metadata---blog-0-b-6-74c",
        },
        Object {
          "content": "content---blog-7-b-8-fd9",
          "metadata": null,
        },
      ],
    },
  },
  "routesConfig": "
import React from 'react';
import ComponentCreator from '@docusaurus/ComponentCreator';
export default [
{
  path: '/blog',
  component: ComponentCreator('/blog','94e'),
  exact: true,
},
{
  path: '*',
  component: ComponentCreator('*')
}
];
",
  "routesPaths": Array [
    "/404.html",
    "/blog",
  ],
}
`;

exports[`loadRoutes nested route config 1`] = `
Object {
  "registry": Object {
    "component---theme-doc-item-178-a40": Object {
      "loader": "() => import(/* webpackChunkName: 'component---theme-doc-item-178-a40' */ \\"@theme/DocItem\\")",
      "modulePath": "@theme/DocItem",
    },
    "component---theme-doc-page-1-be-9be": Object {
      "loader": "() => import(/* webpackChunkName: 'component---theme-doc-page-1-be-9be' */ \\"@theme/DocPage\\")",
      "modulePath": "@theme/DocPage",
    },
    "content---docs-foo-baz-8-ce-61e": Object {
      "loader": "() => import(/* webpackChunkName: 'content---docs-foo-baz-8-ce-61e' */ \\"docs/foo/baz.md\\")",
      "modulePath": "docs/foo/baz.md",
    },
    "content---docs-helloaff-811": Object {
      "loader": "() => import(/* webpackChunkName: 'content---docs-helloaff-811' */ \\"docs/hello.md\\")",
      "modulePath": "docs/hello.md",
    },
    "docsMetadata---docs-routef-34-881": Object {
      "loader": "() => import(/* webpackChunkName: 'docsMetadata---docs-routef-34-881' */ \\"docs-b5f.json\\")",
      "modulePath": "docs-b5f.json",
    },
    "metadata---docs-foo-baz-2-cf-fa7": Object {
      "loader": "() => import(/* webpackChunkName: 'metadata---docs-foo-baz-2-cf-fa7' */ \\"docs-foo-baz-dd9.json\\")",
      "modulePath": "docs-foo-baz-dd9.json",
    },
    "metadata---docs-hello-956-741": Object {
      "loader": "() => import(/* webpackChunkName: 'metadata---docs-hello-956-741' */ \\"docs-hello-da2.json\\")",
      "modulePath": "docs-hello-da2.json",
    },
  },
  "routesChunkNames": Object {
    "/docs/hello-f94": Object {
      "component": "component---theme-doc-item-178-a40",
      "content": "content---docs-helloaff-811",
      "metadata": "metadata---docs-hello-956-741",
    },
    "/docs:route-838": Object {
      "component": "component---theme-doc-page-1-be-9be",
      "docsMetadata": "docsMetadata---docs-routef-34-881",
    },
    "docs/foo/baz-f88": Object {
      "component": "component---theme-doc-item-178-a40",
      "content": "content---docs-foo-baz-8-ce-61e",
      "metadata": "metadata---docs-foo-baz-2-cf-fa7",
    },
  },
  "routesConfig": "
import React from 'react';
import ComponentCreator from '@docusaurus/ComponentCreator';
export default [
{
  path: '/docs:route',
  component: ComponentCreator('/docs:route','838'),
  
  routes: [
{
  path: '/docs/hello',
  component: ComponentCreator('/docs/hello','f94'),
  exact: true,
},
{
  path: 'docs/foo/baz',
  component: ComponentCreator('docs/foo/baz','f88'),
  
},
]
},
{
  path: '*',
  component: ComponentCreator('*')
}
];
",
  "routesPaths": Array [
    "/404.html",
    "/docs/hello",
    "docs/foo/baz",
  ],
}
`;

exports[`loadRoutes route config with empty (but valid) path string 1`] = `
Object {
  "registry": Object {
    "component---hello-world-jse-0-f-b6c": Object {
      "loader": "() => import(/* webpackChunkName: 'component---hello-world-jse-0-f-b6c' */ \\"hello/world.js\\")",
      "modulePath": "hello/world.js",
    },
  },
  "routesChunkNames": Object {
    "-b2a": Object {
      "component": "component---hello-world-jse-0-f-b6c",
    },
  },
  "routesConfig": "
import React from 'react';
import ComponentCreator from '@docusaurus/ComponentCreator';
export default [
{
  path: '',
  component: ComponentCreator('','b2a'),
  
},
{
  path: '*',
  component: ComponentCreator('*')
}
];
",
  "routesPaths": Array [
    "/404.html",
    "",
  ],
}
`;
